pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -705,37 +708,67 @@ func (k *Kubernetes) Undeploy(komposeObject kobject.KomposeObject, opt kobject.C
 		switch t := v.(type) {
 		case *extensions.Deployment:
 			//delete deployment
-			rpDeployment, err := kubectl.ReaperFor(extensions.Kind("Deployment"), client)
+			label := labels.SelectorFromSet(labels.Set(map[string]string{selector: t.Name}))
```

Comment:
Than you can do it like this:
Keep it in for, but move it outside switch.

You can access object metadata fields by casting it to `meta.Object`.
```go
# import "k8s.io/kubernetes/pkg/api/meta"
v.(meta.Object).GetName()
```